test/l2SharedBridge/L2SharedBridgeBase.t.sol

```diff
@@ -0,0 +1,127 @@
+
+contract MintableToken is MockERC20 {
+    function mint(address _to, uint256 _amount) external {
+        _mint(_to, _amount);
+    }
+
+    function burn(uint256 _amount) external {
+        _burn(msg.sender, _amount);
+    }
+}
+
+contract MockL2Messenger {
+    function sendToL1(bytes memory message) external returns (bytes32) {
+        return keccak256(message);
+    }
+}
+
+contract L2SharedBridgeTest is Test {
+    using SafeERC20 for IERC20;
+
+    L2SharedBridge bridge;
+    MintableToken mockL2Token;
+    address l1SharedBridge;
+    address alice;
+    address bob;
+
+    uint256 depositAmount = 100 ether;
+    uint160 constant SYSTEM_CONTRACTS_OFFSET = 0x8000; // 2^15
+
+    function setUp() public {
+        mockL2Token = new MintableToken();
+        mockL2Token.initialize("Mock USDC", "USDC", 18);
+        l1SharedBridge = makeAddr("l1SharedBridge");
+        alice = makeAddr("alice");
+        bob = makeAddr("bob");
+
+        // deploy L2SharedBridge with mock L1 and L2 token addresses
+        address proxyAdmin = makeAddr("proxyAdmin");
+        L2SharedBridge sharedBridgeImpl = new L2SharedBridge(address(mockL2Token), address(mockL2Token));
+        TransparentUpgradeableProxy sharedBridgeProxy = new TransparentUpgradeableProxy(
+            address(sharedBridgeImpl),
+            proxyAdmin,
+            abi.encodeWithSelector(L2SharedBridge.initialize.selector, l1SharedBridge)
+        );
+        bridge = L2SharedBridge(address(sharedBridgeProxy));
+    }
+
+    function testFinalizeDeposit() public {
+        uint256 initialBalance = mockL2Token.balanceOf(bob);
+        address aliasedL1SharedBridge =
+            address(uint160(l1SharedBridge) + uint160(0x1111000000000000000000000000000000001111));
+
+        vm.expectEmit(true, true, true, true);
+        emit IL2SharedBridge.FinalizeDeposit(alice, bob, address(mockL2Token), depositAmount);
+
+        vm.prank(aliasedL1SharedBridge);
+        bridge.finalizeDeposit(alice, bob, address(mockL2Token), depositAmount, "");
+
+        uint256 finalBalance = mockL2Token.balanceOf(bob);
+        assertEq(finalBalance - initialBalance, depositAmount);
+    }
+
+    function testFinalizeDeposit_RevertIfWrongCaller() public {
+        vm.expectRevert(bytes("mq"));
+        bridge.finalizeDeposit(alice, bob, address(mockL2Token), depositAmount, "");
+    }
+
+    function testWithdraw() public {
+        mockL2Token.mint(alice, depositAmount);
+
+        vm.startPrank(alice);
+        mockL2Token.approve(address(bridge), depositAmount);
+
+        // mock L2 messenger call
+        bytes memory expectedMessage =
+            abi.encodePacked(IL1ERC20Bridge.finalizeWithdrawal.selector, bob, address(mockL2Token), depositAmount);
+
+        // initiate withdrawal
+        MockL2Messenger mockMessenger = new MockL2Messenger();
+        vm.etch(address(SYSTEM_CONTRACTS_OFFSET + 0x08), address(mockMessenger).code);
```

Comment:
As previously discussed, this cheat uses `EXTCODECOPY` which is not compatible with zksolc.

Reply:
yep, for the record, will leave as is and we would run tests without --zksync